src/com/lilithsthrone/game/dialogue/utils/OptionsDialogue.java

```diff
@@ -1191,6 +1196,98 @@ private static String getGenerPreferencesPanel(PronounType type) {
 		
 		return sb.toString();
 	}
+
+	private static String getOrientationRepresentation() {
+		float total=0;
+		for(SexualOrientation o : SexualOrientation.values()) {
+			total+=Main.getProperties().orientationPreferencesMap.get(o);
+		}
+		
+		StringBuilder sb = new StringBuilder();
+		
+		if(total==0) {
+			sb.append("<div style='width:100%;height:12px;background:"+Colour.ANDROGYNOUS.toWebHexString()+";float:left;margin:4vw 0 0 0;border-radius: 2px;'>");
+			
+		} else {
+			sb.append("<div style='width:100%;height:12px;background:#222;float:left;margin:4vw 0 0 0;border-radius: 2px;'>");
+			
+			for(SexualOrientation o : SexualOrientation.values()) {
+				sb.append("<div style='width:" + (Main.getProperties().orientationPreferencesMap.get(o)/total) * (100) + "%; height:12px; background:"
+						+ o.getColour().toWebHexString() + "; float:left; border-radius: 2;'></div>");
+			}
+		}
+		
+		sb.append("</div>");
+		
+		return sb.toString();
+	}
+	
+	public static final DialogueNodeOld ORIENTATION_PREFERENCE = new DialogueNodeOld("Orientation preferences", "", true) {
+		private static final long serialVersionUID = 1L;
+		
+		@Override
+		public String getHeaderContent(){
+			UtilText.nodeContentSB.setLength(0);
+			
+			UtilText.nodeContentSB.append(
+					"<div class='container-full-width'>"
+					+ "These options will determine the sexual orientation encounter rates of random NPCs."
+					+ " Note that the race and femininity of NPCs can have an influence on their orientation, but your preferences will be taken into account wherever possible.</br>"
+					+ "<b>A visual representation of the encounter chances can be seen in the bars at the bottom.</b>"
+					+ " (The different shades of each orientation are solely for recognition in the bars, and don't mean anything other than that.)"
+					+ "</div>"
+		
+					+ "<div class='container-full-width' style='text-align:center;'>");
+			
+			UtilText.nodeContentSB.append(getOrientationPreferencesPanel(SexualOrientation.ANDROPHILIC));
+			UtilText.nodeContentSB.append(getOrientationPreferencesPanel(SexualOrientation.AMBIPHILIC));
+			UtilText.nodeContentSB.append(getOrientationPreferencesPanel(SexualOrientation.GYNEPHILIC));
+
+			UtilText.nodeContentSB.append(getOrientationRepresentation() + "</div>");
+			
+			return UtilText.nodeContentSB.toString();
+		}
+		
+		@Override
+		public String getContent(){
+			return "";
+		}
+		
+		@Override
+		public Response getResponse(int responseTab, int index) {
+			 if (index == 0) {
+				return new Response("Back", "Go back to the options menu.", OPTIONS);
+				
+			}else {
```

Comment:
can simply return null; (no else branch necessary - the if branch also returns)

Reply:
There are many ways to express this code, and the decision should be made with readability in mind. (Which, of course, is just opinion. My opinion is either leave as-is, or use a conditional expression across three lines as below.)

```
return index == 0
	? new Response(...)
	: null;
```